tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -875,16 +874,31 @@ public function theAdministratorGivesUserTheRole(string $user, string $role): vo
 	 * @throws Exception
 	 */
 	public function rememberTheAvailableSpaces(): void {
+		$this->availableSpaces = [];
 		$rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
 		$drives = json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR);
 		if (isset($drives["value"])) {
 			$drives = $drives["value"];
 		}
-
 		Assert::assertArrayHasKey(0, $drives, "No drives were found on that endpoint");
 		$spaces = [];
+		$createdUsers = $this->featureContext->getCreatedUsers();
+		$createdUsersId = array_column($createdUsers, 'id');
 		foreach ($drives as $drive) {
-			$spaces[$drive["name"]] = $drive;
+			// when the user gets deleted the spaces related to user isn't deleted so we add only the spaces
+			// of users created by a paticular scenario to the array
+			// this work around can be removed after the fix of https://github.com/owncloud/ocis/issues/4195
+			if ($drive["name"] === 'Admin') {
+				$spaces[$drive["name"]] = $drive;
+			} elseif ($drive["driveType"] === 'personal') {
+				// get the user id of all the personal spaces availabe till now
+				$userId = $drive["owner"]["user"]["id"];
+				if (\in_array($userId, $createdUsersId)) {
+					$spaces[$drive["name"]] = $drive;
+				}
+			} elseif ($drive["driveType"] !== 'personal') {
+				$spaces[$drive["name"]] = $drive;
+			}
```

Comment:
the first and last if blocks are doing the same for different conditions. maybe this can be simplified as:
```suggestion
			// if this case is not for Admin, you can add `$drive["name"] !== 'Admin'` check in combination
			if ($drive["driveType"] === 'personal') {
				$userId = $drive["owner"]["user"]["id"];
				 // we only get the drive created by the user in the current created users list
				if (\in_array($userId, $createdUsersId)) {
					$spaces[$drive["name"]] = $drive;
				}
			} else {
				$spaces[$drive["name"]] = $drive;
			}
```

Reply:
After the changes made by https://github.com/owncloud/ocis/pull/4433 don't think these changes will be necessary